src/internal_events/exec.rs

```diff
@@ -0,0 +1,51 @@
+use super::InternalEvent;
+use metrics::counter;
+
+#[derive(Debug)]
+pub struct ExecEventReceived {
+    pub byte_size: usize,
+}
+
+impl InternalEvent for ExecEventReceived {
+    fn emit_logs(&self) {
+        trace!(message = "Received one event.");
+    }
+
+    fn emit_metrics(&self) {
+        counter!("processed_events_total", 1);
+        counter!("processed_bytes_total", self.byte_size as u64);
+    }
+}
+
+#[derive(Debug)]
+pub struct ExecFailed {
+    pub error: std::io::Error,
+}
+
+impl InternalEvent for ExecFailed {
+    fn emit_logs(&self) {
+        error!(message = "Unable to exec.", error = ?self.error);
+    }
+
+    fn emit_metrics(&self) {
+        counter!("processing_errors_total", 1);
```

Comment:
Could we add `type = "failed"` and `exit_status` here? I think that will help people monitor failed executions.

Reply:
Add the type but not exit_status it was not available for a command spawn error. I have added exit_status to the command_executed_total so I hope that will be a reasonable substitute.